openfasoc/generators/glayout/glayout/flow/blocks/strongARM_US/strongARM_v2.py

```diff
@@ -0,0 +1,221 @@
+    mystrongARM << L_route(pdk, vddpin.ports["e3"] ,mystrongARM.ports["strongARM_clk_reset_pmos_r_welltie_N_top_met_N"], vwidth=1,vglayer="met2",fullbottom=True)
+    
+    #vdd label
+    vddlabel = rectangle(layer=met2_pin, size=(0.1,0.1), centered=True).copy()
+    vddlabel.add_label(text="VDD",layer=met2_label)
+    move_info.append((vddlabel,vddpin.ports["e1"],None))
+    #inp label
+    vinplabel = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vinplabel.add_label(text="VIP",layer=met1_label)
+    move_info.append((vinplabel,mystrongARM.ports["strongARM_diffp_tl_gate_N"],None))
+    #inm label
+    vinmlabel = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vinmlabel.add_label(text="VIM",layer=met1_label)
+    move_info.append((vinmlabel,mystrongARM.ports["strongARM_diffp_tr_gate_N"],None))
+    #vtail label
+    vtaillabel = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vtaillabel.add_label(text="VS",layer=met1_label)
+    move_info.append((vtaillabel,mystrongARM.ports["strongARM_diffp_bl_source_E"],None))
+    #vd1 label
+    vd1label = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vd1label.add_label(text="VP",layer=met1_label)
+    move_info.append((vd1label,mystrongARM.ports["strongARM_diffp_tl_drain_N"],None))
+    #vd2 label
+    vd2label = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vd2label.add_label(text="VQ",layer=met1_label)
+    move_info.append((vd2label,mystrongARM.ports["strongARM_diffp_tr_drain_N"],None))
+    #vclk label
+    vclklabel = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vclklabel.add_label(text="VCLK",layer=met1_label)
+    move_info.append((vclklabel,mystrongARM.ports["strongARM_clk_nmos_gate_N"],None))
+    #gnd label
+    vgndlabel = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vgndlabel.add_label(text="VSS",layer=met1_label)
+    move_info.append((vgndlabel,mystrongARM.ports["strongARM_clk_nmos_tie_N_top_met_S"],None))
+    #vo1 label
+    vo1label = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vo1label.add_label(text="VOP",layer=met1_label)
+    move_info.append((vo1label,mystrongARM.ports["strongARM_cross_couple_top_B_drain_N"],None))
+    #vo2 label
+    vo2label = rectangle(layer=met1_pin, size=(0.1,0.1), centered=True).copy()
+    vo2label.add_label(text="VOM",layer=met1_label)
+    move_info.append((vo2label,mystrongARM.ports["strongARM_cross_couple_top_A_drain_N"],None))
+    #move everything to position
+    for comp, prt, alignment in move_info:
+        alignment = ('c','b') if alignment is None else alignment
+        compref = align_comp_to_port(comp, prt, alignment=alignment)
+        mystrongARM.add(compref)
+    return mystrongARM.flatten()
+
+strong = add_strongARM_labels(sky130_mapped_pdk,mystrongARM(sky130_mapped_pdk, 8, 0.4, 2, 4, 3, 0.4))
```

Comment:
You can add these lines in a separate function or in a separate file so that this block can be imported in other generators without any side effects.